web/markdown/rules.react.js

```diff
@@ -47,8 +71,105 @@ function linkRules(): MarkdownRuleSpec {
 
 // function will contain additional rules for message formatting
 function markdownRules(): MarkdownRuleSpec {
+  const linkMarkdownRules = linkRules();
+
+  const simpleMarkdownRules = {
+    ...linkMarkdownRules.simpleMarkdownRules,
+    autolink: SimpleMarkdown.defaultRules.autolink,
+    link: {
+      ...linkMarkdownRules.simpleMarkdownRules.link,
+      match: SimpleMarkdown.defaultRules.link.match,
+    },
+    blockQuote: {
+      ...SimpleMarkdown.defaultRules.blockQuote,
+      // match end of blockQuote by either \n\n or end of string
+      match: SimpleMarkdown.blockRegex(/^( *>[^\n]+(\n[^\n]+)*\n*)+(\n{2,}|$)/),
+      parse(
+        capture: SimpleMarkdown.Capture,
+        parse: SimpleMarkdown.Parser,
+        state: SimpleMarkdown.State,
+      ) {
+        var content = capture[0].replace(/^ *> ?/gm, '');
+        return {
+          // remove new line from captured string
+          content: parse(content.trim(), state),
+        };
+      },
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => (
+        <blockquote key={state.key} className={css.block}>
+          {output(node.content, state)}
+        </blockquote>
+      ),
+    },
+    inlineCode: {
+      ...SimpleMarkdown.defaultRules.inlineCode,
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => (
+        <code key={state.key} className={css.inlineCode}>
+          {node.content}
+        </code>
+      ),
+    },
+    em: SimpleMarkdown.defaultRules.em,
+    strong: SimpleMarkdown.defaultRules.strong,
+    del: SimpleMarkdown.defaultRules.del,
+    underline: {
+      order: SimpleMarkdown.defaultRules.em.order - 0.5,
+      match: SimpleMarkdown.inlineRegex(/^__([\s\S]+?)__(?!_)/),
```

Comment:
What is the purpose of making the "one or more" quantifier lazy here (`+?`)? I can't figure out what the difference would be with a simple "one or more" quantifier (`+`), since you're using a negative lookahead (`(?!_)`) to prevent following underscores.